language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/documentsymbol/DocumentSymbolResolver.java

```diff
@@ -151,20 +182,184 @@
+                break;
+            default:
+                symbolKind = SymbolKind.TypeParameter;
+        }
+        Range range = DocumentSymbolUtil.generateNodeRange(typeDefinitionNode);
+        Optional<MetadataNode> metadata = typeDefinitionNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
         return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
                 range, range, isDeprecated, children, this.context));
     }
 
+    @Override
+    public Optional<DocumentSymbol> transform(ModuleVariableDeclarationNode moduleVariableDeclarationNode) {
+        BindingPatternNode bindingPatternNode = moduleVariableDeclarationNode.typedBindingPattern().bindingPattern();
+        //Only capture binding pattern is considered. Wildcard and other binding patterns are ignored.
+        String name = bindingPatternNode.kind() == SyntaxKind.CAPTURE_BINDING_PATTERN ?
+                bindingPatternNode.toSourceCode() : "";
+        SymbolKind symbolKind = SymbolKind.Variable;
+        Range range = DocumentSymbolUtil.generateNodeRange(moduleVariableDeclarationNode);
+        Optional<MetadataNode> metadata = moduleVariableDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(ConstantDeclarationNode constantDeclarationNode) {
+        String name = constantDeclarationNode.variableName().text();
+        SymbolKind symbolKind = SymbolKind.Constant;
+        Range range = DocumentSymbolUtil.generateNodeRange(constantDeclarationNode);
+        Optional<MetadataNode> metadata = constantDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(EnumDeclarationNode enumDeclarationNode) {
+        String name = enumDeclarationNode.identifier().text();
+        SymbolKind symbolKind = SymbolKind.Enum;
+        Range range = DocumentSymbolUtil.generateNodeRange(enumDeclarationNode);
+        Optional<MetadataNode> metadata = enumDeclarationNode.metadata();
+        List<DocumentSymbol> children = transformMembers(enumDeclarationNode.enumMemberList());
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
```

Comment:
`createDocumentSymbol` can be a private method. So we don't need to provide the context as an argument.

Reply:
Done